### PR TITLE
Sync user's full name

### PR DESCRIPTION
**Pull request by inlanger**

### CODE REVIEW

File: common/djangoapps/external_auth/backends.py [hunk 1]
```diff
@@ -127,6 +127,10 @@ def authenticate(self, ticket, service):
            NB: Use of PT to identify proxy
         """
         username, authentication_response = _verify(ticket, service)
+
+        first_name = authentication_response[0][1].find(CAS + 'first_name').text
```
Comment (CrowbarKZ) [inline common/djangoapps/external_auth/backends.py]: It's unclear to me how this line works, would be good to add a comment explaining it

Comment (CrowbarKZ) [reply common/djangoapps/external_auth/backends.py]: Also is the response guaranteed to have elements [0] and [1] and so on, or should it be enclose in `try..except`?

File: common/djangoapps/external_auth/views.py [hunk 2]
```diff
@@ -478,13 +478,11 @@ def cas_login(request, next_page=None, required=False):
...
     if request.user.is_authenticated():
         user = request.user
-        UserProfile.objects.get_or_create(
-            user=user,
-            defaults={'name': user.username}
-        )
-        Subscriber.objects.get_or_create(
-            user=user
-        )
+        profile, _ = UserProfile.objects.get_or_create(user=user)
```
Comment (CrowbarKZ) [inline common/djangoapps/external_auth/views.py]: why not keep using `get_or_create(user=user, defaults={'name': user.get_full_name()})`? it seems like 1 less database write with no downsides

File: common/djangoapps/external_auth/views.py [hunk 3]
```diff
@@ -478,13 +478,11 @@ def cas_login(request, next_page=None, required=False):
...
+        profile.name = user.get_full_name()
+        profile.save()
```
Comment (CrowbarKZ) [inline common/djangoapps/external_auth/views.py]: why not keep using `get_or_create(user=user, defaults={'name': user.get_full_name()})`? it seems like 1 less database write with no downsides

**Review by CrowbarKZ — CHANGES_REQUESTED** [review]
left some comments

File: common/djangoapps/external_auth/backends.py (hunk 1 above)
Comment (inlanger) [reply common/djangoapps/external_auth/backends.py]: Here I get user's full name params from cas authentication response(xml tree). 
I've enclosed that part in `try` statement

**Review by aniav — APPROVED** [review]

**Review by CrowbarKZ — APPROVED** [review]

**Pull request merged**